sigstore-java/src/main/java/dev/sigstore/OidcIdentity.java

```diff
@@ -16,25 +16,25 @@
 package dev.sigstore;
 
 import dev.sigstore.oidc.client.OidcToken;
+import org.immutables.value.Value.Default;
 import org.immutables.value.Value.Immutable;
 
 @Immutable
 public interface OidcIdentity {
 
-  static OidcIdentity of(String identity, String issuer) {
+  static OidcIdentity of(StringMatchable identity, StringMatchable issuer) {
     return ImmutableOidcIdentity.builder().identity(identity).issuer(issuer).build();
   }
 
-  static OidcIdentity from(OidcToken oidcToken) {
-    return ImmutableOidcIdentity.builder()
-        .identity(oidcToken.getSubjectAlternativeName())
-        .issuer(oidcToken.getIssuer())
-        .build();
+  @Default
+  default boolean matches(OidcToken oidcToken) {
+    return getIdentity().matches(oidcToken.getSubjectAlternativeName())
+        && getIssuer().matches(oidcToken.getIssuer());
   }
 
   /** The user or machineId. */
-  String getIdentity();
+  StringMatchable getIdentity();
```

Comment:
Is it the same as `SubjectAlternativeName` which already seem to support regexp? https://github.com/sigstore/protobuf-specs/blob/3d01ad8fa60bdd0152bea2963b4d35282c2e263a/protos/sigstore_common.proto#L203-L210

Reply:
Yeah essentially, but doesn't actually implement anything. We still need a native Java datastructure to represent it.

Reply:
protobuf generates a Java representation, doesn't it? We do depend on protobuf anyway, so WDYT of using `SubjectAlternativeName` here?

Reply:
The java representation of protobuf always uses protobuf specific java classes, and I'd rather not expose those in our api.

Reply:
That is fair, however, we seem to expose Bundle, don't we?

Reply:
That's our own representation of bundle. Since the protobuf bundle uses all kinds of proto datastructures